### DISCUSSION TITLE
Get bounce message

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2399] [Help with using Postal]
If a mail bounces, is there any way to get the actual message with webhooks. I have a set up where an employee needs to debug why mails are not delivered and try to fix them. The problem comes when the message is just "Failed 3 times.." or "Undelivered Mail Returned to Sender" with no other context.

For instance I just had a bounce message be due to the recipient not existing. This would have been nice for the person to know directly. Instead I had to manually check the bounce in the postal interface

**Comment by willpower232** [comment]
You can use the message ID to ask the API about the message if you don't already store the message ID when you send the message

https://apiv1.postalserver.io/controllers/messages.html

**Reply by sinnbeck** [reply]
Sounds like a great idea, but sadly it does not seem to contain that information?
```
[
   {
      "id":827,
      "status":"Processed",
      "details":"This has been detected as a bounce message for <msg:700>.",
      "output":null,
      "sent_with_ssl":false,
      "log_id":null,
      "time":null,
      "timestamp":1680526089.621181
   }
]

**Reply by willpower232** [reply]
> You will only receive basic data for this structure, no extended data is available. By default, you'll receive no expansions for the structure. You can add them by providing an array of new expansion names in the _expansions parameter on the request. You can send true rather than array to receive all expansions.

**Reply by sinnbeck** [reply]
That sounds really interesting. I dont assume this is documented anywhere? 

Do you mean that I just add it in the parameters (using the php package)
```
$client->makeRequest('messages', 'deliveries', ['id' => 700, '_expansions' => true]);

**Reply by willpower232** [reply]
I think so
